linkcheck/checker/urlbase.py

```diff
@@ -224,9 +223,7 @@ def set_result(self, msg, valid=True, overwrite=False):
               "Double result %r (previous %r) for %s", msg, self.result, self)
         else:
             self.has_result = True
-        if not isinstance(msg, str_text):
-            log.warn(LOG_CHECK, "Non-unicode result for %s: %r", self, msg)
-        elif not msg:
+        if not msg:
```

Comment:
This part was not obvious to me.  Did a git grep for set_result(), saw that all values passed to it are strings.